apps/server/src/modules/learnroom/service/group-deleted-handler.service.ts

```diff
@@ -0,0 +1,29 @@
+import { Group, GroupDeletedEvent } from '@modules/group';
+import { Injectable } from '@nestjs/common';
+import { EventsHandler, IEventHandler } from '@nestjs/cqrs';
+import { Logger } from '@src/core/logger';
+import { Course, CourseSynchronizationStoppedLoggable } from '../domain';
+import { CourseDoService } from './course-do.service';
+
+@Injectable()
+@EventsHandler(GroupDeletedEvent)
+export class GroupDeletedHandlerService implements IEventHandler<GroupDeletedEvent> {
+	constructor(private readonly courseService: CourseDoService, private readonly logger: Logger) {}
+
+	public async handle(event: GroupDeletedEvent): Promise<void> {
+		await this.removeCourseSyncReference(event.target);
+	}
+
+	private async removeCourseSyncReference(group: Group): Promise<void> {
```

Comment:
I would argue that this belongs in the CourseService. However we have no defined way what Handlers all contain, so we leave it here for now. Will be discussed in ArcGroup.